ocs_ci/ocs/resources/bucket_logging_manager.py

```diff
@@ -0,0 +1,499 @@
+import logging
+import json
+import time
+
+from ocs_ci.framework import config
+from ocs_ci.helpers.helpers import craft_s3_command
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.bucket_utils import list_objects_from_bucket
+from ocs_ci.ocs.exceptions import CommandFailed
+from ocs_ci.utility.utils import TimeoutSampler
+
+
+logger = logging.getLogger(__name__)
+
+
+INTERM_LOGS_PATH = "/var/logs/bucket-logs"
+LOG_CONFG_YAML_PATH = "/spec"
+
+
+class BucketLoggingManager:
+    """
+    This class facilitates MCG bucket logs management
+    """
+
+    cur_logs_pvc = constants.DEFAULT_MCG_BUCKET_LOGS_PVC
+
+    def __init__(self, mcg_obj=None, awscli_pod=None):
+        """
+        Args:
+            mcg_obj(MCG): An MCG object containing required credentials
+            awscli_pod(Pod): A pod for running AWS CLI commands
+        """
+        self.mcg_obj = mcg_obj
+        self.awscli_pod = awscli_pod
+
+    @property
+    def nb_config_resource(self):
+        """
+        Return the NooBaa configuration resource
+        Note that this might change in the future.
+
+        Returns:
+            ocs_ci.ocs.ocp.OCP: OCP instance of the NooBaa configuration resource
+        """
+        return ocp.OCP(
+            kind="noobaa",
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name="noobaa",
+        )
+
+    def enable_bucket_logging_on_cr(self, logs_pvc=None):
+        """
+        Set the guaranteed bucket logs feature
+
+        Args:
+            logs_pvc(str|optional): Name of the bucket logs PVC
+            Note:
+                If not provided, a PVC will be automatically be created
+                by MCG when first enabling the feature.
+        """
+        logger.info("Enabling guaranteed bucket logs")
+
+        # Build a patch command to enable guaranteed bucket logs
+        bucket_logging_dict = {"loggingType": "guaranteed"}
+
+        # Add the bucketLoggingPVC field if provided
+        if logs_pvc:
+            bucket_logging_dict["bucketLoggingPVC"] = logs_pvc
+
+        patch_params = [
+            {
+                "op": "add",
+                "path": f"{LOG_CONFG_YAML_PATH}/bucketLogging",
+                "value": bucket_logging_dict,
+            }
+        ]
+
+        # Try patching via add, and if it fails - replace instead
+        try:
+            self.nb_config_resource.patch(
+                params=json.dumps(patch_params),
+                format_type="json",
+            )
+        except CommandFailed as e:
+            if "already exists" in str(e):
```

Comment:
string.lower_case()

Reply:
Fixed in https://github.com/red-hat-storage/ocs-ci/pull/10228/commits/19cea6f66581ae8941620738a40d4ced08a9470c